### DISCUSSION TITLE
테스트 코드 설계 방안

### DISCUSSION THREAD

**Discussion by jjangsky** [two-clock/backend#120] [General]
@openmpy 님 어느덧 저희가 목표로 하였던 1차 mvp에 도달한 것 같습니다.
다름이 아니라 정식으로 서비스를 배포하기 시작하면 사용자가 존재한다는 가정 하에 다양한 피드백이 올 것이고
그거에 대한 여러 수정 사항들이 생길 것 같습니다.

이러한 이유로 서비스의 안정성을 확보하고자 API에 대해서 테스트 코드를 도입하려고 하는데 
`Persistence`, `Business`, `Presentation` 각 계층에 대해서 모두 도입하기에는 시간적 비용이 아쉬워서
조금 더 우선순위를 정하고 테스트 코드 작성하면 좋지 않을까 싶어 Discussion을 열었습니다.

앞전에 살짝 이야기가 나온 부분이긴 하나 `Business Layer`에서 파라미터 검증과 로직 검증 테스트 코드 작성하고 
repository에 관련된 method들은 stubbing 처리 해두고 `Persistence Layer`에 대해서 실제 DB에 값을 넣어보는 테스트
방식은 어떠한가 의견을 내어봅니다.

**Comment by openmpy** [comment]
@jjangsky 님
시간이 참 빠르네요.. 아주 좋은 제안이십니다.
간단한 CRUD를 제외한 나머지 기능들 위주로 작업 진행하면 될거 같습니다.
일단 우선 저는 회원 관련된 부분 테스트 코드를 작성해보도록 하겠습니다.
@jjangsky 님은 어느 부분 해보고 싶으신가요?

**Reply by jjangsky** [reply]
제가 JPA로 테스트 코드 작성 경험은 적다 보니 연습용으로 간단한 `Comment` 나 `Likes` 관련 도메인부터 해보고 싶네요~

**Reply by openmpy** [reply]
Comment 부분 고고 하시죠
